apps/src/templates/HideToolbarHelper.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import msg from '@cdo/locale';
+import cookies from 'js-cookie';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import trackEvent from '../util/trackEvent';
+import _ from 'lodash';
+
+const styles = {
+  closeX: {
+    position: 'absolute',
+    top: 5,
+    right: 5,
+    fontSize: 13
+  }
+};
+
+// Note that additional styling can be found in apps/style/HideToolbarHelper.scss.
+
+const HidetoolbarHelperCookieName = 'hide_toolbar_helper';
+
+/**
+ * An overlay with instrutions on hiding the toolbar for iPhone with iOS 13.
+ */
+export default class HideToolbarHelper extends React.Component {
+  state = {
+    showHelper: false
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Track whether we knew the toolbar was shown last update, so that
+    // if it disappears, we can track the event in analytics...
+    this.wasToolbarShowing = false;
+
+    // ...but only track it once.
+    this.trackedToolbarHide = false;
+  }
+
+  updateLayout = () => {
+    const isiOS13 = navigator.userAgent.indexOf('iPhone OS 13') !== -1;
+    const isHideCookieSet = cookies.get(HidetoolbarHelperCookieName);
+    const isLandscape = window.orientation !== 0;
+
+    // window.innerHeight is smaller than document.body.offsetHeight when
+    // the iOS 13 Safari toolbar is showing.  It becomes equal when the toolbar
+    // is hidden.  (Interestingly, document.documentElement.clientHeight is
+    // the same as document.body.offsetHeight in both cases.)
+    const isToolbarShowing = window.innerHeight < document.body.offsetHeight;
+
+    // Let's track the disapearance of the toolbar, just once.
+    if (
+      this.wasToolbarShowing &&
+      !isToolbarShowing &&
+      !this.trackedToolbarHide
+    ) {
+      trackEvent('Research', 'HideToolbarHelper', 'hid-' + window.innerHeight);
+
+      this.trackedToolbarHide = true;
+    }
+    this.wasToolbarShowing = isToolbarShowing;
+
+    const showHelper =
+      isiOS13 && !isHideCookieSet && isLandscape && isToolbarShowing;
+
+    this.setState({showHelper});
+  };
+
+  componentDidMount() {
+    this.updateLayout();
+
+    this.updateLayoutListener = _.throttle(this.updateLayout, 200);
+    window.addEventListener('orientationchange', this.updateLayoutListener);
+
+    /* These two events catch all viewport changes. */
+    window.addEventListener('resize', this.updateLayoutListener);
+    window.addEventListener('scroll', this.updateLayoutListener);
```

Comment:
We seem to be adding this set of event listeners in several places around the code base. May be worth moving this to a util function. i.e. `addViewportChangeListener`

Reply:
(same with remove, below)

Reply:
I thought about this over the weekend.  Agreed that an obvious refactor is building more helpers, but I have two concerns.  The first is minor, but it's just that we can end up like [dom.js](https://github.com/code-dot-org/code-dot-org/blob/10f80982f8ca339e8a80a4369e92e03ef44f2041/apps/src/dom.js) with minimal documentation and cases like the [height resizer](https://github.com/code-dot-org/code-dot-org/pull/34185#discussion_r409900596) which seems to only partially use the helpers.  The second is more architectural, and is that I'm not sure the right React way is to have lots of individual components doing their own event management, helper or not.  I think it might be better to centralise this layout management, possibly in [AppView](https://github.com/code-dot-org/code-dot-org/blob/38a67fc9863352b185725b3d470a118d54629236/apps/src/templates/AppView.jsx).  That would be a separate change, however.